---
title: "Microbiome in Alkaline Geothermal Spring At Yellowstone"
author: "Jiucheng Ding"
date: "`r Sys.Date()`"

link-citations: true
output: 
  html_document:
    toc: true
    toc_depth: 2
    toc_float: true
    theme: united
    highlight: tango
editor_options: 
  chunk_output_type: inline
---

Accession: PRJNA756970

Background:
Anoxygenic phototrophs are widely distributed among different habitats and especially in extreme conditions. Factors such as the ability to harvest light and tolerate oxygen are important in determining where phototrophs exist, and studies in hot springs have shown that temperature, pH, and sulfide also play a role. However, previous studies have lacked the necessary resolution to fully understand the diversity of these bacteria (reference).


Introduction:
The samples were collected for the purpose of a grant. The name of the project was "Phototropy in eruptive hot springs". The project aims to discover the distribution and activity of metabolically diverse anoxygenic phototrophs across geothermal spings. 4 sample were collected for this project, from 2 locations of sampling (FC,JJ) and with each has two different temperatures (hot and cool). 

This R Markdown analysis aims to discover the role of temperature that may play in affecting the diversity and composition of the microbiome from the sample. 


#Before this analysis, file "preprocessed_physeq" has already been generated and the .RData file can be found in data folder.


#Here we are going to load the packages and sources required. 
```{r load-packages}
# Efficiently load packages 
pacman::p_load(phyloseq, iNEXT, ggpubr, vegan, tidyverse, install = FALSE)

# load in functions and color preferences
source("/local/workdir/jd2263/BIOMI6300_Project2_AmpliconAnalysis/code/functions.R")
source("/local/workdir/jd2263/BIOMI6300_Project2_AmpliconAnalysis/code/colors_and_shapes.R")
```


# The following chunk will produce the rarefaction plot so that we can have a brief idea of how the species richness increase as the number of sequences increase. Interestingly, we can see that both 0324 and 0326 (the cool samples) have plateaued at a lower richness than 0325 and 0327. This could be suggesting that there might be a relationship between higher temperature and lower species richness.


# Load data
```{r load-data_preprocessed_physeq}
load("/local/workdir/jd2263/BIOMI6300_Project2_AmpliconAnalysis/data/preprocessed_physeq.RData")
preprocessed_physeq
```

## Work with the iNEXT package 
```{r iNEXT-div-estimation}
# Prepare data for iNEXT
iNEXT_input_df <- 
  preprocessed_physeq %>%
  otu_table() %>%
  data.frame()

# Run iNEXT on the data   
# Warning: Takes ~5 mins to run!
# Calculate diversity with iNEXT
set.seed(777)
# WARNING: This command takes a while 
# Calculating over night 
iNEXT_data <- iNEXT(iNEXT_input_df, q = c(0, 1, 2), datatype = "abundance")
save(iNEXT_data, file = "/local/workdir/jd2263/BIOMI6300_Project2_AmpliconAnalysis/data/iNEXT_Yellowstone.RData")

load("/local/workdir/jd2263/BIOMI6300_Project2_AmpliconAnalysis/data/iNEXT_Yellowstone.RData")
# Note that the object is called "iNEXT_data"

typeof(iNEXT_data)

# Make workable dataframe ##################### The following dataframe must be fixed.
div_iNEXT_df <- 
  iNEXT_data$AsyEst %>%
  dplyr::rename(names = Assemblage) %>%
  make_MA_metadata()
```

```{r rarefaction}
location_colors
# Set colors for manual plotting with iNEXT

background_col_FL <- "goldenrod1"
background_col_PA <- "firebrick3"
background_col_WH <- "darkorange2"

# Prepare data to color the figure for iNEXT
dat <- colnames(iNEXT_input_df) %>%  
  data.frame() 
colnames(dat)[1] <- "names"     
sub_metadata <- dat %>% make_MA2_metadata()
# Add the colors for the plotting!
dat_iNEXT <- dat %>%    
  left_join(sub_metadata, by = "names") %>%  
  mutate(location_color = ifelse(location == "FC", background_col_PA, 
                                 ifelse(location == "JJ", background_col_FL)))
                                
                
# Now plot it all :) 
# Plot the rarefaction/extrapolation curve (type = 1)
rarefaction_fixed_plot <- 
  ggiNEXT(iNEXT_data, type=1, facet.var="Order.q") + 
  facet_wrap(~Order.q, scales="fixed") +
  #scale_color_manual(values = dat_iNEXT$location_color,  guide = FALSE) +
  #scale_fill_manual(values = dat_iNEXT$location_color, guide = FALSE) +
  theme(legend.position = "right",axis.text.x = element_text(size = 8)) + 
    labs(x = "Number of Sequences", y="Species Richness")
rarefaction_fixed_plot
# Save the file! 
ggsave(rarefaction_fixed_plot, 
       filename = "figures/rarefaction_fixed.png",
       units = "in", height = 3.5, width = 8)
```


# The following two chunks will check the sample size distribution and visualize it.
```{r load-data}
load("/local/workdir/jd2263/BIOMI6300_Project2_AmpliconAnalysis/data/preprocessed_physeq.RData")
preprocessed_physeq
# What are the sample size distributions?
preprocessed_physeq %>%
  sample_sums() %>%
  data.frame()  
  #View()

# Remove sample that has fewest number of reads
# The fewest reads is 24489 and the next was 25951

physeq_2267 <- preprocessed_physeq %>%
  prune_samples(sample_sums(.) > 2000, .)
```

```{r scale_reads}
set.seed(777) 
# Scale the reads using function scale_reads from functions.R 
# Intution check of number of sequeces per sample
min(sample_sums(physeq_2267))
scaled_physeq2267 <-  
  physeq_2267 %>%
  scale_reads(round = "matround")
  
# Now check the sequencing depth! 
scaled_seqSums_df <-   
scaled_physeq2267 %>%  
otu_table(.) %>%  # Take the sum of the sample abundance   
colSums(.) %>%  
data.frame() %>%  
rownames_to_column(., var = "names") %>% 
 make_MA_metadata()
 
 # Rename the 2nd column 
 colnames(scaled_seqSums_df)[2] <- "TotalSeqs"   
 
scaled_seqSums_df %>%  
  ggplot(aes(x=TotalSeqs)) +
  ggtitle("Histogram of Raw Read Counts Samples") +  
  geom_histogram(bins = 40) + 
  theme(legend.position = c(0.15, 0.85)) +   
  scale_y_continuous(expand = c(0, 0), limits = c(0, 4)) +   
  scale_x_continuous(limits = c(24480, 24490), breaks = seq(24480,24490,5))

```
This figure simply shows the sample sequence size distribution at this step. We can tell that all 4 of the samples are within a very small range, between 24480 and 24490.


# PCoA, plot the PCoA showing the Bray-Curtis disimilarities of the data:
```{r BC}
# Calculate Bray-Curtis Dissimilarity 
scaled_BC_pcoa <- 
  ordinate(
    physeq = scaled_physeq2267,
    method = "PCoA",
    distance = "bray", 
    binary = FALSE
  )


plot_ordination(
  physeq = scaled_physeq2267,
  ordination = scaled_BC_pcoa,
  color = "heat",
  shape = "location", 
  title = "Scaled Bray-Curtis PCoA"
) + 
  geom_point(
    size = 5, 
    alpha = 6, 
    aes(color = heat, shape = location), 
    stroke = 0.8, 
    show.legend = TRUE
  ) +
  scale_color_manual(values = heat_colors, guide = guide_legend(override.aes = list(shape = 15))) +
  scale_shape_manual(values = location_shapes)

```
From the PCoA, we can visualize that there is a big cap caused by the difference in color. Although the sample size is very small, we can still sense that there must be significant diversity variation caused by different temperature.




# Stats!

# Check for differences in the communities using PERMANOVA!


```{r permanova}
# Calculate bray curtis for PERMANOVA
scaled_bray <- phyloseq::distance(scaled_physeq2267, method = "bray", binary = FALSE)
# pull out metadata 

metadata <- scaled_physeq2267 %>%
  sample_data() %>%
  data.frame()
# Permutational Multivariate Analysis of Variance Using Distance Matrices
# aka PERMANOVA using the adonis2 function from vegan 

# using the bray curtis 
# Testing if the centroids of the fractions are different? 
adonis2(scaled_bray ~ location, data = metadata)

adonis2(scaled_bray ~ heat, data = metadata)

# Are they independent of each other?
adonis2(scaled_bray ~ location +heat,data = metadata)
# Is there an interaction between the variables? 
adonis2(scaled_bray ~ location * heat, data = metadata)
```
From this PERMANOVA test, we can get conclusion that:
The first test shows the results of the adonis2 test with only location as a predictor, which indicates that location does not significantly explain the variation in the data (p-value = 0.6667).

The second test shows the results of the adonis2 test with only heat as a predictor, which also indicates that heat does not significantly explain the variation in the data (p-value = 0.3333).

The third test shows the results of the adonis2 test with both location and heat as predictors, which also indicates that neither location nor heat significantly explain the variation in the data (p-values = 0.5000 and 0.1667, respectively).

The fourth test shows the results of the adonis2 test with an interaction term between location and heat as a predictor, which shows that the model explains all of the variation in the data and that there is no residual variation left (R2 = 1).

However, the results may also be due to the fact that the sample size is extremely small. If resources allow, the sample size should be much bigger and it will be easier to generate significant conclusion from it.



# Homogeniety of dispersion test: Beta dispr

Dispersion between heat(hot/cool): and between location(Fc/JJ):

```{r beta dispr}
#Between different heats(hot/cool):
betadispr_heat <- betadisper(scaled_bray, metadata$heat)
permutest(betadispr_heat)

#Between locations (FC, JJ):
betadispr_location<- betadisper(scaled_bray, metadata$location)

permutest(betadispr_location)
```

The output from permutest indicates that there is a significant difference in multivariate dispersion between the two groups ("cold" and "heat"). The p-value is 0.04167, which is less than 0.05 (assuming alpha=0.05). The ANOVA F-test also suggests a significant difference between groups, with a very large F-statistic and an essentially perfect fit (which may make the summary unreliable).




Then perform phylum compositional analysis by using a stack boxplot:
# Compositional Analysis

``` {r}
phylum_df <- 
  scaled_physeq2267 %>%
  tax_glom(taxrank = "Phylum") %>%                     # Agglomerate/merge counts by phylum 
  transform_sample_counts(function(x) {x/sum(x)}) %>%  # Transform to relative abundances
  psmelt() %>%                                         # melt into long format data frame 
  dplyr::filter(Abundance > 0.01)                   # filter out phyla that are > 1% abundant 
  

# Group the data by Phylum and Sample, and calculate the total Abundance for each group
phylum_df_grouped <- phylum_df %>%
  group_by(Phylum, Sample) %>%
  summarize(total_abundance = sum(Abundance))

# Create the stacked bar chart
ggplot(data = phylum_df_grouped, aes(x = Sample, y = total_abundance, fill = Phylum)) +
  geom_col() +
  labs(x = "Sample", y = "Total Abundance", fill = "Phylum") +
  theme_bw()
```


The output of the phylum distribution shows that Chloroflexi is the most dominant phylum mostly across each sample. Interestingly, we can see that it has a lower composition in cooler condition than hotter condition. Could temperature directly or indirectly have an interplay with the compositional distribution of these anoxygenic microbiome? This is a very interesting topic to delve into.

Conclusion:
The analysis presented in this R Markdown document aims to explore the role of temperature on the diversity and composition of the microbiome from samples collected from two different locations with varying temperatures. The data was preprocessed and rarefaction plots were created to visualize the species richness. Interestingly, the cool samples plateaued at a lower richness than the hot samples, which could suggest a relationship between higher temperature and higher species richness. The sample size distributions were also checked and visualized.

However, this analysis has some limitations. Firstly, the small sample size of only four samples from two locations may not be representative of the entire population of anoxygenic phototrophs across geothermal springs. Secondly, the analysis only focuses on the role of temperature and does not take into account other environmental factors that could affect the diversity and composition of the microbiome. Overall, further research and analysis with a larger dataset and a more comprehensive approach would be needed to confirm the findings presented in this document.

# Session Information 
```{r}
# Reproducibility
devtools::session_info()
```
