---
title: "Compositional Analysis"
author: "Jiucheng Ding"
date: "`r Sys.Date()`"
output: 
  html_document:
    toc: true
    toc_depth: 2
    toc_float: true
    theme: united
    highlight: tango
editor_options: 
  chunk_output_type: inline
---

Accession: PRJNA756970

Background




Introduction:

The samples were collected for the purpose of a grant. The name of the project was "Phototropy in eruptive hot springs". The project aims to discover the distribution and activity of metabolically diverse anoxygenic phototrophs across geothermal spings. 4 sample were collected for this project, from 2 locations of sampling (FC,JJ) and with each has two different temperatures (hot and cool). 
This R Markdown analysis aims to discover the role of temperature that may play in affecting the diversity and composition of the microbiome from the sample. 






Here we are going to load the packages and sources required. \# Load packages

```{r load-packages}
# Efficiently load packages 
pacman::p_load(phyloseq, iNEXT, ggpubr, vegan, tidyverse, install = FALSE)

# load in functions and color preferences
source("/local/workdir/jd2263/BIOMI6300_Project2_AmpliconAnalysis/code/functions.R")
source("/local/workdir/jd2263/BIOMI6300_Project2_AmpliconAnalysis/code/colors_and_shapes.R")
```

# Load data

```{r load-data}
load("/local/workdir/jd2263/BIOMI6300_Project2_AmpliconAnalysis/data/preprocessed_physeq.RData")
preprocessed_physeq
# What are the sample size distributions?
preprocessed_physeq %>%
  sample_sums() %>%
  data.frame()  %>%
  View()

# Remove sample that has fewest number of reads
# The fewest reads is 1839 and the next was 2267.
# So, I decided to use those extra 400 reads for more accurate beta div analysis
physeq_2267 <- preprocessed_physeq %>%
  prune_samples(sample_sums(.) > 2000, .)
```

```{r scale_reads}
set.seed(777) 
# Scale the reads using function scale_reads from functions.R 
# Intution check of number of sequeces per sample
min(sample_sums(physeq_2267))
scaled_physeq2267 <-  
  physeq_2267 %>%
  scale_reads(round = "matround")
  
# Now check the sequencing depth! 
scaled_seqSums_df <-   
scaled_physeq2267 %>%  
otu_table(.) %>%  # Take the sum of the sample abundance   
colSums(.) %>%  
data.frame() %>%  
rownames_to_column(., var = "names") %>% 
 make_MA_metadata()
 
 # Rename the 2nd column 
 colnames(scaled_seqSums_df)[2] <- "TotalSeqs"   
 
scaled_seqSums_df %>%  
  ggplot(aes(x=TotalSeqs)) +
  ggtitle("Histogram of Raw Read Counts Samples") +  
  geom_histogram(bins = 40) + 
  theme(legend.position = c(0.15, 0.85)) +   
  scale_y_continuous(expand = c(0, 0), limits = c(0, 4)) +   
  scale_x_continuous(limits = c(24480, 24490), breaks = seq(24480,24490,5))

```

# PCoA

```{r BC}
# Calculate Bray-Curtis Dissimilarity 
scaled_BC_pcoa <- 
  ordinate(
    physeq = scaled_physeq2267,
    method = "PCoA",
    distance = "bray", 
    binary = FALSE
  )
# Plot the PCoA 
plot_ordination(
  physeq = scaled_physeq2267,
  ordination = scaled_BC_pcoa,
  color = "location",
  shape = "location", 
  title = "Scaled Bray-Curtis PCoA") + 
  geom_point(size = 5, alpha = 0.5, aes(color = location)) 
  scale_color_manual(values = location_colors)

# heat 
plot_ordination(
  physeq = scaled_physeq2267,
  ordination = scaled_BC_pcoa,
  color = "heat",
  shape = "heat", 
  title = "Scaled Bray-Curtis PCoA") + 
  geom_point(size = 5, alpha = 0.5, aes(color = heat))  
  scale_color_manual(values = heat_colors)

```

# Stats!

# Check for differences in the communities using PERMANOVA

```{r permanova}
# Calculate bray curtis for PERMANOVA
scaled_bray <- phyloseq::distance(scaled_physeq2267, method = "bray", binary = FALSE)
# pull out metadata 

metadata <- scaled_physeq2267 %>%
  sample_data() %>%
  data.frame()
# Permutational Multivariate Analysis of Variance Using Distance Matrices
# aka PERMANOVA using the adonis2 function from vegan 
# Test the hypothesis that the fraction centriods are different 
# using the bray curtis 
# Testing if the centroids of the fractions are different? 
adonis2(scaled_bray ~ location, data = metadata)
# Are the centroids different for stations based on bray curtis?
adonis2(scaled_bray ~ heat, data = metadata)
adonis2(scaled_bray ~ pH, data = metadata)
# Are they independent of each other?
adonis2(scaled_bray ~ location +heat+ pH, data = metadata)
# Is there an interaction between the variables? 
adonis2(scaled_bray ~ location * heat * pH, data = metadata)
```

# Homogeniety of dispersion test: Beta dispr

Dispersion between heat(hot/cool): and between location(Fc/JJ):

```{r}
#Between different heats:
betadispr_heat <- betadisper(scaled_bray, my_metadata$heat)
permutest(betadispr_heat)

#Between locations (FC, JJ):
betadispr_location<- betadisper(scaled_bray, my_metadata$location)
permutest(betadispr_location)


```

The output from permutest indicates that there is a significant difference in multivariate dispersion between the two groups ("cold" and "heat"). The p-value is 0.04167, which is less than 0.05 (assuming alpha=0.05). The ANOVA F-test also suggests a significant difference between groups, with a very large F-statistic and an essentially perfect fit (which may make the summary unreliable).

Then perform phylum compositional analysis by using a stack boxplot:

# Compositional Analysis

``` {r}
phylum_df <- 
  scaled_physeq2267 %>%
  tax_glom(taxrank = "Phylum") %>%                     # Agglomerate/merge counts by phylum 
  transform_sample_counts(function(x) {x/sum(x)}) %>%  # Transform to relative abundances
  psmelt() %>%                                         # melt into long format data frame 
  dplyr::filter(Abundance > 0.01)                   # filter out phyla that are > 1% abundant 
  

# Group the data by Phylum and Sample, and calculate the total Abundance for each group
phylum_df_grouped <- phylum_df %>%
  group_by(Phylum, Sample) %>%
  summarize(total_abundance = sum(Abundance))

# Create the stacked bar chart
ggplot(data = phylum_df_grouped, aes(x = Sample, y = total_abundance, fill = Phylum)) +
  geom_col() +
  labs(x = "Sample", y = "Total Abundance", fill = "Phylum") +
  theme_bw()
```
